core/src/main/java/com/linecorp/armeria/server/annotation/JavaBeanRequestConverterFunction.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation;
+
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.convertParameter;
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.httpParametersOf;
+import static org.reflections.ReflectionUtils.getAllFields;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpParameters;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.util.Holder;
+import com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.EnumConverter;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.util.AsciiString;
+
+public class JavaBeanRequestConverterFunction implements RequestConverterFunction {
+    private static final Map<Class<?>, BeanParamInfo> beanParamsMap = new HashMap<>();
```

Comment:
is it need to be concurrent? See that same logic at JsonConverter need ConcurrentHashmap, so there is a chance that convertRequest will be called concurrently right?

Reply:
fixed.
Changed to `ConcurrentMap` type, and `beanParamInfoCache` field name